src/analyses/dependence_graph.cpp

```diff
@@ -114,14 +107,46 @@ void dep_graph_domaint::control_dependencies(
         post_dom_all=false;
     }
 
-    if(post_dom_all ||
-       !post_dom_one)
-      control_deps.erase(it);
-
-    it=next;
+    if(post_dom_all || !post_dom_one)
+    {
+      control_deps.erase(cd);
+    }
+    else
+    {
+      control_deps.insert(cd);
+    }
   }
 }
 
+bool dep_graph_domaint::merge_control_dependencies(
+  const depst &other_control_deps,
```

Comment:
To reduce the cognitive load, it would be great if all `deps` could be renamed to `dependencies`